### PR TITLE
individual application lookup via angular service

### PR DESCRIPTION
**Pull request by jlaswell**
Dashboard now provides listing of applications. Application details are available at '/app/applications/:name'. Styling changes.

screen shot 2016-12-15 at 10 58 42 am
screen shot 2016-12-15 at 10 59 08 am

### CODE REVIEW

**Review by kyleferguson — CHANGES_REQUESTED** [review]
This is all looking really good! What do you think about referencing apps by a UUID? Or at the very least AI ID. Better queries down the road, not subject to character issues, etc. Especially when it comes to REST URLs and user provided names

**Comment by jlaswell** [comment]
@kyleferguson I'd prefer a UUID myself. I'll add the migration and update the routes!

File: app/routes/rootRoutes.go [hunk 1]
```diff
@@ -41,7 +40,7 @@ func BuildRouter() *negroni.Negroni {
 
 	api := apiRoutes()
 	r.SubRoute("/api/v1", negroni.New(
-		negroni.HandlerFunc(auth.APIMiddleware),
+		// negroni.HandlerFunc(auth.APIMiddleware),
```
Comment (jlaswell) [inline app/routes/rootRoutes.go]: silly

File: app/handlers/api/applications.go [hunk 2]
```diff
@@ -31,3 +32,25 @@ func Applications(w http.ResponseWriter, r *http.Request) {
...
+		http.Error(w, "there was an error when attempting to connect to the database", http.StatusInternalServerError)
+		return
+	}
+	id := bone.GetValue(r, "uuid")
+	app, err := resources.GetApplication(db, id)
+	if err != nil {
+		log.Printf("Error when retrieving application: '%s'\n", err)
+		http.Error(w, "there was an error when retrieving application", http.StatusInternalServerError)
+		return
+	}
```
Comment (kyleferguson) [inline app/handlers/api/applications.go:51]: I think eventually we should distinguish the difference between an unrecoverable error, and something like a 404 not found, but I'm fine with merging this as is and looking at later.

File: app/handlers/api/applications.go [hunk 3]
```diff
@@ -31,3 +32,25 @@ func Applications(w http.ResponseWriter, r *http.Request) {
 	}
 	json.NewEncoder(w).Encode(apps)
 }
+
+// Application provides an Application
+func Application(w http.ResponseWriter, r *http.Request) {
```
Comment (kyleferguson) [inline app/handlers/api/applications.go]: Can we standardize the basic REST structure handlers as `[Resource][Method]`, where method would be show, index, store, etc. So we would have `api.ApplicationIndex` and `api.ApplicationShow`. What are your thoughts on that?

**Review by kyleferguson — CHANGES_REQUESTED** [review]

File: app/handlers/api/applications.go (hunk 2 above)
Comment (jlaswell) [reply app/handlers/api/applications.go:51]: I'm in agreement there.

File: app/handlers/api/applications.go (hunk 3 above)
Comment (jlaswell) [reply app/handlers/api/applications.go]: I think that's a better way to handle this. I didn't know how to feel about the classic `Index`, `Show`, `Create` wording, but I think that it's best for this case.

**Comment by jlaswell** [comment]
@kyleferguson updated!

**Review by kyleferguson — APPROVED** [review]

**Pull request merged**